src/main/java/data/AnalysisResults.java

```diff
@@ -36,4 +39,37 @@ public String getRawText() {
   public ImmutableSet<String> getCategories() {
     return categories;
   }
-}
\ No newline at end of file
+
+  public Optional<String> getStore() {
+    return store;
+  }
+
+  public static class Builder {
+    private final String rawText;
+    private ImmutableSet<String> categories;
+    private Optional<String> store;
+
+    public Builder(String rawText) {
+      this.rawText = rawText;
+    }
+
+    public String getRawText() {
+      return rawText;
+    }
+
+    public Builder setCategories(Set<String> categories) {
+      this.categories = ImmutableSet.copyOf(categories);
+      return this;
+    }
+
+    public Builder setStore(String store) {
+      this.store = Optional.of(store);
+      return this;
+    }
+
+    public AnalysisResults build() {
+      store = Optional.ofNullable(store).map(value -> value.get());
```

Comment:
Does `store = Optional.ofNullable(store);` not work?

Reply:
`store` is of type `Optional<String>`, so that would make a `Optional<Optional<String>>`.

Reply:
Should I instead make the `store` property of the Builder a `String`, and then create the `Optional` in the `build()` method using `ofNullable`?

Reply:
I see. So when do we expect that store would be null?

Reply:
For the case where no logo is detected, `setStore()` would not be called and the `AnalysisResults` would have the store set to an empty `Optional`.

Reply:
Oh, can you default it to empty: `private Optional<String> store = Optional.empty();`